lib/packages/core/src/api/factories/account/setRewardRecipient.ts

```diff
@@ -0,0 +1,50 @@
+/** @module core.api.factories */
+
+/**
+ * Copyright (c) 2019 Burst Apps Team
+ */
+import { BurstService } from '../../../service/burstService';
+import { TransactionId } from '../../../typings/transactionId';
+import { TransactionResponse } from '../../../typings/transactionResponse';
+import { generateSignature } from '@burstjs/crypto';
+import { verifySignature } from '@burstjs/crypto';
+import { generateSignedTransactionBytes } from '@burstjs/crypto';
+import { convertNumberToNQTString } from '@burstjs/util';
+import { broadcastTransaction } from '../transaction/broadcastTransaction';
+
+/**
+ * Use with [[ApiComposer]] and belongs to [[AccountApi]].
+ *
+ * See details at [[AccountApi.setRewardRecipient]]
+ */
+export const setRewardRecipient = (service: BurstService): (
+        recipient: string,
+        feeNQT: string,
+        senderPublicKey: string,
+        senderPrivateKey: string,
+        deadline: number,
+    ) => Promise<TransactionId> =>
+    async (
+        recipient: string,
+        feeNQT: string,
+        senderPublicKey: string,
+        senderPrivateKey: string,
+        deadline: number,
+    ): Promise<TransactionId> => {
+
+        let parameters = {
+            recipient,
+            deadline: 1440,
+            feeNQT: convertNumberToNQTString(parseFloat(feeNQT)),
+            publicKey: senderPublicKey
+        };
+        const {unsignedTransactionBytes: unsignedHexMessage} = await service.send<TransactionResponse>('setRewardRecipient', parameters);
+        const signature = generateSignature(unsignedHexMessage, senderPrivateKey);
```

Comment:
here we do always the same... I need to extract this part....too much boilerplate